### PR TITLE
Add support for AV1 encoding/decoding to FFmpeg and mpv

### PR DESCRIPTION
**Pull request by hyiltiz**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by javian** [comment]
Thank you for the contribution @hyiltiz and your first PR to Homebrew. This PR however is not going to be accepted since despite you putting the work in the AOM formula is based on a head only repository. We need to have a properly released version of the software before it can make it in to Homebrew.

**Comment by hyiltiz** [comment]
As you know, some excellent software (I can name a few, but as I said, I am assuming you already know a few) simply does not have a "release" per se, and simply works as a rolling release. In fact, most companies now do, calling it SaaS. Therefore, strictly requiring a "release" tag seems mute.

**Comment by javian** [comment]
Yes I'm aware =) but lets not get into a debate about that here. As described in the https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md document standalone formulas that are using head only is not acceptable. 

If you submit the ffmpeg formula with av1 as a resource it should be potentially possible to include it but someone else in the team would need to pitch in on that.

File: Formula/aom.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Aom < Formula
+  desc "Alliance for Open Media Codec SDK"
+  homepage "http://aomedia.org"
+  head "https://aomedia.googlesource.com/aom.git"
```
Comment (fxcoudert) [inline Formula/aom.rb]: We can't accept this. I see that there is a `v1.0.0` tag which, if it is considered “stable” by upstream, we can use as a tagged version.

File: Formula/aom.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    mkdir "aom-build" do
+      args = std_cmake_args
+      args << ".."
+      system "cmake", *args
+      system "make", "install"
+    end
+  end
+
+  test do
+    system ./aomenc
```
Comment (fxcoudert) [inline Formula/aom.rb]: We need a test that exercises the some of the functionality of the app.

File: Formula/ffmpeg.rb [hunk 3]
```diff
@@ -1,17 +1,17 @@
 class Ffmpeg < Formula
   desc "Play, record, convert, and stream audio and video"
   homepage "https://ffmpeg.org/"
-  url "https://ffmpeg.org/releases/ffmpeg-4.0.2.tar.xz"
-  sha256 "a95c0cc9eb990e94031d2183f2e6e444cc61c99f6f182d1575c433d62afb2f97"
+  url "https://ffmpeg.org/releases/ffmpeg-3.4.2.tar.bz2"
```
Comment (fxcoudert) [inline Formula/ffmpeg.rb]: No decreasing version numbers!

File: Formula/ffmpeg.rb [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
   head "https://github.com/FFmpeg/FFmpeg.git"
 
   bottle do
-    sha256 "83f8f5c5b79b7bbe6e13ab918f4bd8effb1eba2e7091a5776eeb31db84e74102" => :mojave
-    sha256 "5522a58020c3b5c68e5628818ee746822fe38747810c29b4aa865f4fe2803d57" => :high_sierra
-    sha256 "98cdb186d1dd9c9eb2149b5d5d5c15dc13cecb58fed25232abcbeeb1236c92b8" => :sierra
-    sha256 "4ef0ba0bd512fff69004bb33417a8516d1d8e9dc5e3eb05ea9f2ac15f557cdb2" => :el_capitan
+    sha256 "9df7c5868b825944edc80822e818028b4123ee8cdf85ef60eb11fed235b23c79" => :high_sierra
+    sha256 "18b50630056fbe4a3d12515f32728242349b43bd7b956f3401190861b2ccd402" => :sierra
+    sha256 "e507900d3ec72618b365588223f57428654334119d34d6d61ca58210bb7c4907" => :el_capitan
```
Comment (fxcoudert) [inline Formula/ffmpeg.rb]: Do not touch the bottle blocks please.

File: Formula/ffmpeg.rb [hunk 5]
```diff
@@ -106,9 +105,11 @@ def install
       --host-ldflags=#{ENV.ldflags}
     ]
 
+    args << "--disable-jack" if build.stable?
```
Comment (fxcoudert) [inline Formula/ffmpeg.rb]: Why disabling jack?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
I have left review comments on the pull request code itself. We will not be shipping this into Homebrew unless a stable release is tagged. Using it as an unversioned resource is also not acceptable.

**Comment by hyiltiz** [comment]
Maybe @duvrai might be interested to help out here.

**Comment by zmwangx** [comment]
Interestingly, I was compiling FFmpeg with libaom myself just now, then I noticed this PR.

In fact, `aom.git` does have a `v1.0.0` tag, tagged in June:

```git
tag v1.0.0
Tagger:     Yaowu Xu <yaowu@google.com>
TaggerDate: Mon Jun 25 10:14:26 2018 -0700

AOMedia Codec Workgroup Approved version 1.0

commit d14c5bb4f336ef1842046089849dee4a301fbbf0 (HEAD, tag: v1.0.0)
Author:     Yaowu Xu <yaowu@google.com>
AuthorDate: Mon Jun 25 07:54:59 2018 -0700
Commit:     Yaowu Xu <yaowu@google.com>
CommitDate: Mon Jun 25 07:54:59 2018 -0700

    [NORMATIVE]Allow non-identical sequence headers

    This commit changes the reference code to allow additional sequence
    headers with different number of trailing 0 bits.

    Change-Id: Icbb1c9a834b438271456fcda9323a3212dd62784
```

Not sure if it wasn't pushed to origin at the time of your discussion.

My trivial `libaom.rb`:

```rb
class Libaom < Formula
  desc "Alliance for Open Media AV1 codec"
  homepage "https://aomedia.googlesource.com/aom/"
  url "https://aomedia.googlesource.com/aom.git",
      :tag => "v1.0.0"

  depends_on "cmake" => :build

  def install
    mkdir "build" do
      system "cmake", "..", *std_cmake_args
      system "make", "install"
    end
  end

  test do
    raise "Not implemented."
  end
end
```

**Comment by hyiltiz** [comment]
Yes, there was! And that is an interesting find. 
https://aomedia.googlesource.com/aom/+/v1.0.0

**Pull request closed without merging**